components/forms/conditionOfJoining/CojView.tsx

```diff
@@ -11,23 +11,32 @@ import { acceptanceValidation } from "../formr-part-b/ValidationSchema";
 import { Redirect } from "react-router-dom";
 import { updatedsigningCoj } from "../../../redux/slices/userSlice";
 import { COJ_DECLARATIONS } from "../../../utilities/Constants";
+import { DateUtilities } from "../../../utilities/DateUtilities";
 
 const CojView: React.FC = () => {
   const signingCoj = store.getState().user.signingCoj;
   const progName = store.getState().user.signingCojProgName;
+  const signedDate = store.getState().user.signingCojSignedDate;
+  const canEdit = signedDate === null;
```

Comment:
This canEdit implementation does do the job but might be worth looking at again e.g. you already have the signedDate as a prop so if the canEdit is dependent on signedDate then why not just use the signedDate/!signedDate as the condition? (apologies if there's an obvious reason for not doing this! 😁)